feat: configurable roll-forward of flex grading

nsprenkle opened the pull request:
## Description

Add ability to roll-forward ORA flex peer grading feature.

When **enabled** for an Organization or course, flex peer grading will be turned on at the course level for new course reruns.

When **disabled** for an Organization or course, flex peer grading setting will be **unmodified** from default (for new course) or previous (for rerun) value.
 
## Supporting information

JIRA: https://2u-internal.atlassian.net/browse/AU-1289

## Testing instructions

**Regression behavior:**
1. Create a rerun of a course using example API requests.
2. The **FEATURE NAME TBD** value on the course should be false.

**Negative case:**
1. Create a new Organization override in [Home](http://localhost:18000/admin/) › [Waffle_Utils](http://localhost:18000/admin/waffle_utils/) › [Waffle flag org overrides](http://localhost:18000/admin/waffle_utils/waffleflagorgoverridemodel/) called `openresponseassessment.force_on_flexible_peer_grading`. Set to **FORCE OFF**.
3. Create a rerun of a course using example API requests.
4. The **FEATURE NAME TBD** value on the course should be **FALSE**.

**New behavior:**
1. Create/edit an Organization override in [Home](http://localhost:18000/admin/) › [Waffle_Utils](http://localhost:18000/admin/waffle_utils/) › [Waffle flag org overrides](http://localhost:18000/admin/waffle_utils/waffleflagorgoverridemodel/) called `openresponseassessment.force_on_flexible_peer_grading`. Set to **FORCE ON**.
3. Create a rerun of a course using example API requests.
4. The **FEATURE NAME TBD** value on the course should be **TRUE**.

## Other information

**Dependencies:**
- [x] Work in [AU-1269](https://2u-internal.atlassian.net/browse/AU-1269) to establish the field in course metadata.

nsprenkle committed:
feat: configurable roll-forward of flex grading

Add ability to roll-forward ORA flex peer grading feature. When enabled
for an Organization, flex peer grading will be turned on at the course
level for new course reruns.

nsprenkle committed:
refactor: refactor flex grading check in toggles

nsprenkle committed:
test: test flex grading on rerun toggle

jansenk commented on cms/djangoapps/contentstore/views/course.py:
```diff
@@ -61,6 +61,7 @@
 from openedx.core.djangoapps.credit.tasks import update_credit_course_requirements
 from openedx.core.djangoapps.models.course_details import CourseDetails
 from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.waffle_utils import CourseWaffleFlag
```
remove import

jansenk commented on cms/djangoapps/contentstore/views/course.py:
```diff
@@ -1038,6 +1042,9 @@ def rerun_course(user, source_course_key, org, number, run, fields, background=T
 
+    # Enable certain fields rolling forward, where configured
+    fields['force_on_flexible_peer_openassessments'] = enable_flexible_peer_openassessments_on_rerun(source_course_key)
+
```
If the previous run had turned the course setting on but the org has the wafffle off, this would force the setting off. I don't think that's what we want.

```suggestion
    if enable_flexible_peer_openassessments_on_rerun(source_course_key):
    fields['force_on_flexible_peer_openassessments'] = True 
```

nsprenkle replied:
Ah, so basically only rolling-forward the feature, never toggling it off? I could buy that.

nsprenkle replied:
@jansenk , this should be updated, welcoming re-review

nsprenkle committed:
feat: enable flex grading on new course toggle

When a new course is created, flex grading for ORAs will be enabled by
default where instructed by the feature toggle.

nsprenkle committed:
test: update flex grading on rerun toggle tests

nsprenkle committed:
feat: update flext toggle behavior to only toggle on, not off

nsprenkle committed:
chore: remove unused import

nsprenkle committed:
style: remove trailing whitespace

jansenk commented on cms/djangoapps/contentstore/toggles.py:
```diff
@@ -476,3 +476,26 @@ def use_new_course_team_page(course_key):
+def default_enable_flexible_peer_openassessments(course_key):
+    """
+    Returns a boolean if ORA flexible peer grading should be toggled on for a
+    course rerun. We expect this to be set at the organization level to opt
```
or new course

jansenk commented on cms/djangoapps/contentstore/tests/test_course_create_rerun.py:
```diff
@@ -317,3 +318,54 @@ def test_course_creation_without_permission_for_specific_organization(self):
+            'description': 'Test roll-forward of flex grading setting',
+        })
+        source_course = self.store.get_course(self.source_course_key)
+        source_course.force_on_flexible_peer_openassessments = mock_original_course_setting
```
it's the source_course of course!

jansenk commented on cms/djangoapps/contentstore/tests/test_course_create_rerun.py:
```diff
@@ -317,3 +318,54 @@ def test_course_creation_without_permission_for_specific_organization(self):
+    @mock.patch(
+        'cms.djangoapps.contentstore.views.course.default_enable_flexible_peer_openassessments'
+    )
+    def test_default_enable_flexible_peer_openassessments_on_rerun(
```
great test 👍

nsprenkle committed:
style: fix pylint issue

nsprenkle committed:
docs: update toggle comments
